src/mudlet.cpp

```diff
@@ -714,246 +418,611 @@ mudlet::mudlet()
+                currentTranslation.mNativeName = QStringLiteral("English (British)");
+            } else if (!languageCode.compare(QLatin1String("zh_CN"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("简化字");
+            } else if (!languageCode.compare(QLatin1String("zh_TW"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("繁體字");
+            } else if (!languageCode.compare(QLatin1String("nl_NL"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("Nederlands");
+            } else if (!languageCode.compare(QLatin1String("fr_FR"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("Français");
+            } else if (!languageCode.compare(QLatin1String("de_DE"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("Deutsch");
+            } else if (!languageCode.compare(QLatin1String("el_GR"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("ελληνικά");
+            } else if (!languageCode.compare(QLatin1String("it_IT"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("Italiano");
+            } else if (!languageCode.compare(QLatin1String("pl_PL"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("Polszczyzna");
+            } else if (!languageCode.compare(QLatin1String("ru_RU"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("Pусский");
+            } else if (!languageCode.compare(QLatin1String("es_ES"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("Español");
+            } else if (!languageCode.compare(QLatin1String("pt_PT"), Qt::CaseInsensitive)) {
+                currentTranslation.mNativeName = QStringLiteral("Portugês");
+            } else {
+                currentTranslation.mNativeName = languageCode;
+            }
+            currentTranslation.mMudletTranslationFileName = translationFileName;
+            mTranslationsMap.insert(languageCode, currentTranslation);
+        } else {
+            // This is very unlikely to be reached as it means that a file that
+            // matched the naming to be a Mudlet translation file was not infact
+            // one...
+            qDebug().noquote().nospace() << "    no Mudlet translation found for locale code: \"" << languageCode << "\".";
+        }
+        // As pQtTranslator goes out of scope at the end of each loop it will
+        // dispose of the temporarily created QTranslator for us...
+    }
+}
+
+// To be used AFTER scanForMudletTranslations(...) has been called, this will
+// insert the corresponding Qt system translation pathFileNames (path to
+// filenames: qt_xx.qm or qt_xx_YY.qm) into the entries in the mTranslationsMap
+// QMap.
```

Comment:
> This looks like code smell...

Not sure I understand what you mean by that. :confused:

This method works on `(QMap<QString, translation>) mudlet::mTranslationsMap` to add the details for a suitable `Qt library translation` file into each `translation` class instance once those have been populated with one for each Mudlet translation.  There might be a different way to sequence this (after determining **each** Mudlet translation we *then* go and find a matching Qt translation *if there is one* for that locale before considering the next Mudlet translation) but there does not seem to be a compelling reason why this *has* to be done that way instead...

Reply:
Call the method from within the method it depends on with the data it needs as arguments, what do you think? A bit of functional design there.

Reply:
I created `scanForMudletTranslations` so it could be used to load any .qm files needed - this includes Qt's. What is the big difference between scanForMudletTranslations and `scanForQtTranslations`? None really, this new function largely duplicates what the other function does, just does it in a slightly different way. I think this is a bad design.